src/keyboard.rs

```diff
@@ -0,0 +1,1667 @@
+                FavoriteRecall2, FavoriteRecall3, FavoriteStore0, FavoriteStore1, FavoriteStore2,
+                FavoriteStore3, Guide, GuideNextDay, GuidePreviousDay, Info, InstantReplay, Link,
+                ListProgram, LiveContent, Lock, MediaApps, MediaAudioTrack, MediaLast,
+                MediaSkipBackward, MediaSkipForward, MediaStepBackward, MediaStepForward,
+                MediaTopMenu, NavigateIn, NavigateNext, NavigateOut, NavigatePrevious,
+                NextFavoriteChannel, NextUserProfile, OnDemand, Pairing, PinPDown, PinPMove,
+                PinPToggle, PinPUp, PlaySpeedDown, PlaySpeedReset, PlaySpeedUp, RandomToggle,
+                RcLowBattery, RecordSpeedNext, RfBypass, ScanChannelsToggle, ScreenModeNext,
+                Settings, SplitScreenToggle, STBInput, STBPower, Subtitle, Teletext, VideoModeNext,
+                Wink, ZoomToggle, F1, F2, F3, F4, F5, F6, F7, F8, F9, F10, F11, F12, F13, F14, F15,
+                F16, F17, F18, F19, F20, F21, F22, F23, F24, F25, F26, F27, F28, F29, F30, F31,
+                F32, F33, F34, F35
+            }
+        )
+    }
+}
+
+impl Key {
+    /// Convert a key to its approximate textual equivalent.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use winit::keyboard::Key;
+    ///
+    /// assert_eq!(Key::Character("a".into()).to_text(), Some("a"));
+    /// assert_eq!(Key::Enter.to_text(), Some("\r"));
+    /// assert_eq!(Key::F20.to_text(), None);
+    /// ```
+    pub fn to_text(&self) -> Option<&str> {
+        match self {
+            Key::Character(ch) => Some(ch.as_str()),
+            Key::Enter => Some("\r"),
+            Key::Backspace => Some("\x08"),
+            Key::Tab => Some("\t"),
+            Key::Space => Some(" "),
+            Key::Escape => Some("\x1b"),
+            _ => None,
+        }
+    }
+}
+
+/// The location of the key on the keyboard.
+///
+/// Certain physical keys on the keyboard can have the same value, but are in different locations.
+/// For instance, the Shift key can be on the left or right side of the keyboard, or the number
+/// keys can be above the letters or on the numpad. This enum allows the user to differentiate
+/// them.
+///
+/// See the documentation for the [`location`] field on the [`KeyEvent`] struct for more information.
+///
+/// [`location`]: ../event/struct.KeyEvent.html#structfield.location
+/// [`KeyEvent`]: crate::event::KeyEvent
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum KeyLocation {
+    /// The key is in its "normal" location on the keyboard.
+    ///
+    /// For instance, the "1" key above the "Q" key on a QWERTY keyboard will use this location. This
+    /// invariant is also returned when the location of the key cannot be identified.
+    ///
+    /// ![Standard 1 key](https://raw.githubusercontent.com/rust-windowing/winit/master/images/keyboard_standard_1_key.svg)
```

Comment:
Since there's no neat solution for rust-lang/rust#32104, I guess linking to it through GitHub works fine for now. My gut tells me we should be linking to a tagged release rather than the `master` branch, but linking to `master` initially is fine. I did find a way to link to the local copy of the SVG first, and then fall back to the GitHub link (roughly `<img src="../../../../images/keyboard_standard_1_key.svg" onerror="this.src = 'https://raw.githubusercontent.com/rust-windowing/winit/master/images/keyboard_standard_1_key.svg'">`), but that wouldn't link the local version of the image if Winit's docs are built as a dependency, though I suppose instances where that would matter are exceedingly rare, if nonexistent.